h2o-algos/src/main/java/hex/ensemble/StackedEnsembleModel.java

```diff
@@ -205,10 +225,16 @@ boolean isUsefulBaseModel(Key<Model> baseModelKey) {
   }
 
   ModelMetrics doScoreMetricsOneFrame(Frame frame, Job job) {
-      Frame pred = this.predictScoreImpl(frame, new Frame(frame), null, job, true, CFuncRef.from(_parms._custom_metric_func));
-//      Frame pred = this.score(frame, null, job, true,  CFuncRef.from(_parms._custom_metric_func));
+    Frame scoredFrame = (_parms._score_training_samples > 0 && _parms._score_training_samples < frame.numRows())
+            ? MRUtils.sampleFrame(frame, _parms._score_training_samples, _parms._seed)
```

Comment:
Sampling will oftentimes be more expensive than using zero weights to eliminate unwanted rows.

Reply:
@michalkurka Good to know... Do you know what we do in Deep Learning where we also score on just a sub-sample the training set?

Reply:
In Deep Learning, we use the sampleFrame method (`hex/deeplearning/DeepLearning.java:378`), that's where I found it.

Reply:
@michalkurka @sebhrusen I am trying to find a way to use zero weights, so far it seems to me that I have to create a new column with weights and then for each base model (and meta learner) change their property `_weights_column` and protected property `_hasWeights` and then score everything and finally return the properties to their original values. (I am not sure if that's safe)

I guess we can have some cached version of a model on each node so I should do the `write_lock`, `update`, `unlock`, right?

Or is there some better way to do it?

Reply:
agree that it looks fishy, would not recommend it.
@michalkurka any existing example showing how to do this safely? Or we can consider this as an improvement for a separate PR...

Reply:
I think given that we are making such a small sample (10k) rows any optimization doesn't make much sense. I didn't realize the sample is so "petite". Thanks for looking into it though.